### DISCUSSION TITLE
crouton is now EOL

### DISCUSSION THREAD

**Discussion by dnschneid** [dnschneid/crouton#4654] [General]
**edited 2024-12-31**

This hopefully won't come as a surprise to anyone (especially if you've looked recently at the [code frequency graphs](https://github.com/dnschneid/crouton/graphs/code-frequency)), but crouton has been seeing little to no feature development for quite a while now.  As I have recently switched jobs, it's as good a time as ever to be a bit more intentional about the future of crouton.  More details below.

It's been a fantastic ~9~ [12 years](https://github.com/dnschneid/crouton/commit/eff65a0d007728bf23bc52ee9f2927523758befb), and could not have happened without the help of @drinkcat, @smibarber, and all of you who stepped up to be [contributors](https://github.com/dnschneid/crouton/blob/master/CONTRIBUTORS).  Special thanks to @DennisLfromGA who has been a phenomenal moderator and guide for the community.  And considerable thanks to all of you who have been helping each other out in the issue tracker, on Reddit, and on random blog posts/videos/cat gifs across the interblogospherimetaweb3.  Without you all, crouton could not have grown to the extent it did (keep in mind I have never advertised crouton anywhere!), and perhaps crostini in turn might not have come to be.

Speaking of crostini, while crouton still has its niches (especially with EOL devices), [crostini](https://support.google.com/chromebook/answer/9145439) development has long since surpassed crouton for convenience and performance on a majority of our use cases.  I would encourage everyone to try it out and see if it works for you.  If you really want a full Linux desktop environment, you can of course continue to use crouton or one of the many other (some very recent!) methods of installing common Linux distros.  And if you do decide to continue to use crouton (thanks!), the issue tracker, discussion area, and wiki here on Github will continue to be open for business.

So without further ado (and this has been added to the README [mostly verbatim](https://github.com/dnschneid/crouton/blob/master/README.md#crouton-is-now-maintenance-only)):

### 🪦 crouton is now end-of-life 🪦

This means that:
 * The repo is now locked, and no further changes will be considered.
 * Eventually someone will want the latest Ubuntu added to the release list. See [this commit](https://github.com/dnschneid/crouton/commit/6d80f57b91c39d10b29fde861aac5a2b5b9b3910) for an example of how to do it on your own copy.
 * Sometime around July 2025, the GitHub project will be archived, making the issue tracker, discussions, and wiki read-only.
 * For the safety of users and stability of crouton's functionality for those on EOL devices, offers to take over the dnschneid/crouton repo or extension will be declined, and requests to change link destinations will be rejected.

**Comment by joebobbio** [comment]
This is sad to hear, I hope an alternative crops up soon

**Comment by Paumanok** [comment]
This project was invaluable for me working through my degree back in 2014. Sad to see it go but my chromebook has long been out of support range so its bare metal now.

Thanks for all the work you put in!

**Comment by ghost** [comment]
Thanks for the work you put into this project, I used it a lot about four years ago. Good luck with your future plans!

**Comment by pjchamberlain** [comment]
Thank you for Crouton. It is sad but understandable and I confess I've not needed it since Crostini became available to me.

**Comment by cvmiller** [comment]
I just want to chime in with the others, that Crouton has made ChromeOS usable in ways that Google has never considered, or will support. The ability to run applications directly on Linux-base OS, has been invaluable for me. Specifically running `wicd` for wireless troubleshooting, and IPv6 stack debugging, two items which cannot be done in Crostini (because of the VM/Container)

Even the Crostini Devs give credit to Crouton. 

https://chromium.googlesource.com/chromiumos/docs/+/master/containers_and_vms.md#Why-the-name-Crostini

Clearly the Google Team realized that Linux was important, and created a usable instance of Linux, which is now supported by them. Crostini doesn't do everything, but it does most, and the integration with ChromeOS is pretty good.

Thanks for all your years of support, and creating a project which fills in the gaps of ChromeOS

**Comment by ShipmasterKyle** [comment]
Crouton will be missed, hopefully alternatives popup soon.

**Comment by ilovegabagoolsomuch** [comment]
I've made an alternative project (with 70+ stars). [Here](http://milky.software) are the docs for it :D. It got hundreds of upvotes on reddit, though you do need a Linux PC to build the bootable USB. You can find the Github page [here](https://github.com/milkydeveloper/cb-linux).

**Reply by TheTechRobo** [reply]
Is the website down? I can,t load it.

**Reply by ilovegabagoolsomuch** [reply]
Fixed! Try clicking the link again.

**Reply by TheTechRobo** [reply]
Yep!

**Reply by cvmiller** [reply]
Do you see Breath supporting ARM at some point in the future?

**Comment by CroutonIsFun** [comment]
Crouton is still the best all-around method for having a full Linux desktop. All the others have limitations. Chromebrew is pretty good, but there is no way to install .deb packages using dpkg. Crostini is very nice, but without access to Chromebook hardware, you can't play/rip CDs or DVDs.

Regarding Crouton, the debian release bullseye is very nice and ubuntu focal, though officially "unsupported", works quite well. Meanwhile, jammy appears to be broken, and I'm not referring to sound issues: I was not able to initiate an x-session. I think xiwi works, but I'm still testing stuff.

Other solutions require you to dual boot, and this is no fun. Crouton is a work of pure genius, comparable in my opinion to Bach's Toccota and Fugue in D minor, which he wrote when he was 20.
